// @axelar-network/axelarjs-types/build/commonjs/axelar/permission/v1beta1/tx.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeregisterControllerResponse = exports.DeregisterControllerRequest = exports.RegisterControllerResponse = exports.RegisterControllerRequest = exports.UpdateGovernanceKeyResponse = exports.UpdateGovernanceKeyRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const keys_1 = require("../../../cosmos/crypto/multisig/keys");
exports.protobufPackage = "axelar.permission.v1beta1";
function createBaseUpdateGovernanceKeyRequest() {
    return { sender: new Uint8Array(), governanceKey: undefined };
}
exports.UpdateGovernanceKeyRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.governanceKey !== undefined) {
            keys_1.LegacyAminoPubKey.encode(message.governanceKey, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGovernanceKeyRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.governanceKey = keys_1.LegacyAminoPubKey.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            governanceKey: isSet(object.governanceKey) ? keys_1.LegacyAminoPubKey.fromJSON(object.governanceKey) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.governanceKey !== undefined &&
            (obj.governanceKey = message.governanceKey ? keys_1.LegacyAminoPubKey.toJSON(message.governanceKey) : undefined);
        return obj;
    },
    create(base) {
        return exports.UpdateGovernanceKeyRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateGovernanceKeyRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.governanceKey = (object.governanceKey !== undefined && object.governanceKey !== null)
            ? keys_1.LegacyAminoPubKey.fromPartial(object.governanceKey)
            : undefined;
        return message;
    },
};
function createBaseUpdateGovernanceKeyResponse() {
    return {};
}
exports.UpdateGovernanceKeyResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGovernanceKeyResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.UpdateGovernanceKeyResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseUpdateGovernanceKeyResponse();
        return message;
    },
};
function createBaseRegisterControllerRequest() {
    return { sender: new Uint8Array(), controller: new Uint8Array() };
}
exports.RegisterControllerRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.controller.length !== 0) {
            writer.uint32(18).bytes(message.controller);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterControllerRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.controller = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            controller: isSet(object.controller) ? bytesFromBase64(object.controller) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.controller !== undefined &&
            (obj.controller = base64FromBytes(message.controller !== undefined ? message.controller : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.RegisterControllerRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterControllerRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.controller = object.controller ?? new Uint8Array();
        return message;
    },
};
function createBaseRegisterControllerResponse() {
    return {};
}
exports.RegisterControllerResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterControllerResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.RegisterControllerResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseRegisterControllerResponse();
        return message;
    },
};
function createBaseDeregisterControllerRequest() {
    return { sender: new Uint8Array(), controller: new Uint8Array() };
}
exports.DeregisterControllerRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.controller.length !== 0) {
            writer.uint32(18).bytes(message.controller);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeregisterControllerRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.controller = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            controller: isSet(object.controller) ? bytesFromBase64(object.controller) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.controller !== undefined &&
            (obj.controller = base64FromBytes(message.controller !== undefined ? message.controller : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.DeregisterControllerRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeregisterControllerRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.controller = object.controller ?? new Uint8Array();
        return message;
    },
};
function createBaseDeregisterControllerResponse() {
    return {};
}
exports.DeregisterControllerResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeregisterControllerResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.DeregisterControllerResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseDeregisterControllerResponse();
        return message;
    },
};
var tsProtoGlobalThis = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (tsProtoGlobalThis.Buffer) {
        return Uint8Array.from(tsProtoGlobalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = tsProtoGlobalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (tsProtoGlobalThis.Buffer) {
        return tsProtoGlobalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(String.fromCharCode(byte));
        });
        return tsProtoGlobalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
