// @axelar-network/axelarjs-types/build/commonjs/axelar/multisig/exported/v1beta1/types.js
"use strict";
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.keyStateToJSON = exports.keyStateFromJSON = exports.KeyState = exports.multisigStateToJSON = exports.multisigStateFromJSON = exports.MultisigState = exports.protobufPackage = void 0;
exports.protobufPackage = "axelar.multisig.exported.v1beta1";
var MultisigState;
(function (MultisigState) {
    MultisigState[MultisigState["MULTISIG_STATE_UNSPECIFIED"] = 0] = "MULTISIG_STATE_UNSPECIFIED";
    MultisigState[MultisigState["MULTISIG_STATE_PENDING"] = 1] = "MULTISIG_STATE_PENDING";
    MultisigState[MultisigState["MULTISIG_STATE_COMPLETED"] = 2] = "MULTISIG_STATE_COMPLETED";
    MultisigState[MultisigState["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(MultisigState = exports.MultisigState || (exports.MultisigState = {}));
function multisigStateFromJSON(object) {
    switch (object) {
        case 0:
        case "MULTISIG_STATE_UNSPECIFIED":
            return MultisigState.MULTISIG_STATE_UNSPECIFIED;
        case 1:
        case "MULTISIG_STATE_PENDING":
            return MultisigState.MULTISIG_STATE_PENDING;
        case 2:
        case "MULTISIG_STATE_COMPLETED":
            return MultisigState.MULTISIG_STATE_COMPLETED;
        case -1:
        case "UNRECOGNIZED":
        default:
            return MultisigState.UNRECOGNIZED;
    }
}
exports.multisigStateFromJSON = multisigStateFromJSON;
function multisigStateToJSON(object) {
    switch (object) {
        case MultisigState.MULTISIG_STATE_UNSPECIFIED:
            return "MULTISIG_STATE_UNSPECIFIED";
        case MultisigState.MULTISIG_STATE_PENDING:
            return "MULTISIG_STATE_PENDING";
        case MultisigState.MULTISIG_STATE_COMPLETED:
            return "MULTISIG_STATE_COMPLETED";
        case MultisigState.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.multisigStateToJSON = multisigStateToJSON;
var KeyState;
(function (KeyState) {
    KeyState[KeyState["KEY_STATE_UNSPECIFIED"] = 0] = "KEY_STATE_UNSPECIFIED";
    KeyState[KeyState["KEY_STATE_ASSIGNED"] = 1] = "KEY_STATE_ASSIGNED";
    KeyState[KeyState["KEY_STATE_ACTIVE"] = 2] = "KEY_STATE_ACTIVE";
    KeyState[KeyState["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(KeyState = exports.KeyState || (exports.KeyState = {}));
function keyStateFromJSON(object) {
    switch (object) {
        case 0:
        case "KEY_STATE_UNSPECIFIED":
            return KeyState.KEY_STATE_UNSPECIFIED;
        case 1:
        case "KEY_STATE_ASSIGNED":
            return KeyState.KEY_STATE_ASSIGNED;
        case 2:
        case "KEY_STATE_ACTIVE":
            return KeyState.KEY_STATE_ACTIVE;
        case -1:
        case "UNRECOGNIZED":
        default:
            return KeyState.UNRECOGNIZED;
    }
}
exports.keyStateFromJSON = keyStateFromJSON;
function keyStateToJSON(object) {
    switch (object) {
        case KeyState.KEY_STATE_UNSPECIFIED:
            return "KEY_STATE_UNSPECIFIED";
        case KeyState.KEY_STATE_ASSIGNED:
            return "KEY_STATE_ASSIGNED";
        case KeyState.KEY_STATE_ACTIVE:
            return "KEY_STATE_ACTIVE";
        case KeyState.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.keyStateToJSON = keyStateToJSON;
