// @axelar-network/axelarjs-types/build/commonjs/axelar/multisig/v1beta1/genesis.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const params_1 = require("./params");
const types_1 = require("./types");
exports.protobufPackage = "axelar.multisig.v1beta1";
function createBaseGenesisState() {
    return { params: undefined, keygenSessions: [], signingSessions: [], keys: [], keyEpochs: [] };
}
exports.GenesisState = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.params !== undefined) {
            params_1.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.keygenSessions) {
            types_1.KeygenSession.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.signingSessions) {
            types_1.SigningSession.encode(v, writer.uint32(26).fork()).ldelim();
        }
        for (const v of message.keys) {
            types_1.Key.encode(v, writer.uint32(34).fork()).ldelim();
        }
        for (const v of message.keyEpochs) {
            types_1.KeyEpoch.encode(v, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.params = params_1.Params.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.keygenSessions.push(types_1.KeygenSession.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.signingSessions.push(types_1.SigningSession.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.keys.push(types_1.Key.decode(reader, reader.uint32()));
                    break;
                case 5:
                    message.keyEpochs.push(types_1.KeyEpoch.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: isSet(object.params) ? params_1.Params.fromJSON(object.params) : undefined,
            keygenSessions: Array.isArray(object?.keygenSessions)
                ? object.keygenSessions.map((e) => types_1.KeygenSession.fromJSON(e))
                : [],
            signingSessions: Array.isArray(object?.signingSessions)
                ? object.signingSessions.map((e) => types_1.SigningSession.fromJSON(e))
                : [],
            keys: Array.isArray(object?.keys) ? object.keys.map((e) => types_1.Key.fromJSON(e)) : [],
            keyEpochs: Array.isArray(object?.keyEpochs) ? object.keyEpochs.map((e) => types_1.KeyEpoch.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.params !== undefined && (obj.params = message.params ? params_1.Params.toJSON(message.params) : undefined);
        if (message.keygenSessions) {
            obj.keygenSessions = message.keygenSessions.map((e) => e ? types_1.KeygenSession.toJSON(e) : undefined);
        }
        else {
            obj.keygenSessions = [];
        }
        if (message.signingSessions) {
            obj.signingSessions = message.signingSessions.map((e) => e ? types_1.SigningSession.toJSON(e) : undefined);
        }
        else {
            obj.signingSessions = [];
        }
        if (message.keys) {
            obj.keys = message.keys.map((e) => e ? types_1.Key.toJSON(e) : undefined);
        }
        else {
            obj.keys = [];
        }
        if (message.keyEpochs) {
            obj.keyEpochs = message.keyEpochs.map((e) => e ? types_1.KeyEpoch.toJSON(e) : undefined);
        }
        else {
            obj.keyEpochs = [];
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.params = (object.params !== undefined && object.params !== null)
            ? params_1.Params.fromPartial(object.params)
            : undefined;
        message.keygenSessions = object.keygenSessions?.map((e) => types_1.KeygenSession.fromPartial(e)) || [];
        message.signingSessions = object.signingSessions?.map((e) => types_1.SigningSession.fromPartial(e)) || [];
        message.keys = object.keys?.map((e) => types_1.Key.fromPartial(e)) || [];
        message.keyEpochs = object.keyEpochs?.map((e) => types_1.KeyEpoch.fromPartial(e)) || [];
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
