// @axelar-network/axelarjs-types/build/module/axelar/multisig/v1beta1/params.js
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { Threshold } from "../../utils/v1beta1/threshold";
export const protobufPackage = "axelar.multisig.v1beta1";
function createBaseParams() {
    return {
        keygenThreshold: undefined,
        signingThreshold: undefined,
        keygenTimeout: Long.ZERO,
        keygenGracePeriod: Long.ZERO,
        signingTimeout: Long.ZERO,
        signingGracePeriod: Long.ZERO,
        activeEpochCount: Long.UZERO,
    };
}
export const Params = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keygenThreshold !== undefined) {
            Threshold.encode(message.keygenThreshold, writer.uint32(10).fork()).ldelim();
        }
        if (message.signingThreshold !== undefined) {
            Threshold.encode(message.signingThreshold, writer.uint32(18).fork()).ldelim();
        }
        if (!message.keygenTimeout.isZero()) {
            writer.uint32(24).int64(message.keygenTimeout);
        }
        if (!message.keygenGracePeriod.isZero()) {
            writer.uint32(32).int64(message.keygenGracePeriod);
        }
        if (!message.signingTimeout.isZero()) {
            writer.uint32(40).int64(message.signingTimeout);
        }
        if (!message.signingGracePeriod.isZero()) {
            writer.uint32(48).int64(message.signingGracePeriod);
        }
        if (!message.activeEpochCount.isZero()) {
            writer.uint32(56).uint64(message.activeEpochCount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.keygenThreshold = Threshold.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.signingThreshold = Threshold.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.keygenTimeout = reader.int64();
                    break;
                case 4:
                    message.keygenGracePeriod = reader.int64();
                    break;
                case 5:
                    message.signingTimeout = reader.int64();
                    break;
                case 6:
                    message.signingGracePeriod = reader.int64();
                    break;
                case 7:
                    message.activeEpochCount = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            keygenThreshold: isSet(object.keygenThreshold) ? Threshold.fromJSON(object.keygenThreshold) : undefined,
            signingThreshold: isSet(object.signingThreshold) ? Threshold.fromJSON(object.signingThreshold) : undefined,
            keygenTimeout: isSet(object.keygenTimeout) ? Long.fromValue(object.keygenTimeout) : Long.ZERO,
            keygenGracePeriod: isSet(object.keygenGracePeriod) ? Long.fromValue(object.keygenGracePeriod) : Long.ZERO,
            signingTimeout: isSet(object.signingTimeout) ? Long.fromValue(object.signingTimeout) : Long.ZERO,
            signingGracePeriod: isSet(object.signingGracePeriod) ? Long.fromValue(object.signingGracePeriod) : Long.ZERO,
            activeEpochCount: isSet(object.activeEpochCount) ? Long.fromValue(object.activeEpochCount) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        message.keygenThreshold !== undefined &&
            (obj.keygenThreshold = message.keygenThreshold ? Threshold.toJSON(message.keygenThreshold) : undefined);
        message.signingThreshold !== undefined &&
            (obj.signingThreshold = message.signingThreshold ? Threshold.toJSON(message.signingThreshold) : undefined);
        message.keygenTimeout !== undefined && (obj.keygenTimeout = (message.keygenTimeout || Long.ZERO).toString());
        message.keygenGracePeriod !== undefined &&
            (obj.keygenGracePeriod = (message.keygenGracePeriod || Long.ZERO).toString());
        message.signingTimeout !== undefined && (obj.signingTimeout = (message.signingTimeout || Long.ZERO).toString());
        message.signingGracePeriod !== undefined &&
            (obj.signingGracePeriod = (message.signingGracePeriod || Long.ZERO).toString());
        message.activeEpochCount !== undefined &&
            (obj.activeEpochCount = (message.activeEpochCount || Long.UZERO).toString());
        return obj;
    },
    create(base) {
        return Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.keygenThreshold = (object.keygenThreshold !== undefined && object.keygenThreshold !== null)
            ? Threshold.fromPartial(object.keygenThreshold)
            : undefined;
        message.signingThreshold = (object.signingThreshold !== undefined && object.signingThreshold !== null)
            ? Threshold.fromPartial(object.signingThreshold)
            : undefined;
        message.keygenTimeout = (object.keygenTimeout !== undefined && object.keygenTimeout !== null)
            ? Long.fromValue(object.keygenTimeout)
            : Long.ZERO;
        message.keygenGracePeriod = (object.keygenGracePeriod !== undefined && object.keygenGracePeriod !== null)
            ? Long.fromValue(object.keygenGracePeriod)
            : Long.ZERO;
        message.signingTimeout = (object.signingTimeout !== undefined && object.signingTimeout !== null)
            ? Long.fromValue(object.signingTimeout)
            : Long.ZERO;
        message.signingGracePeriod = (object.signingGracePeriod !== undefined && object.signingGracePeriod !== null)
            ? Long.fromValue(object.signingGracePeriod)
            : Long.ZERO;
        message.activeEpochCount = (object.activeEpochCount !== undefined && object.activeEpochCount !== null)
            ? Long.fromValue(object.activeEpochCount)
            : Long.UZERO;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
