// @axelar-network/axelarjs-types/build/commonjs/axelar/snapshot/v1beta1/params.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "axelar.snapshot.v1beta1";
function createBaseParams() {
    return { minProxyBalance: long_1.default.ZERO };
}
exports.Params = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.minProxyBalance.isZero()) {
            writer.uint32(8).int64(message.minProxyBalance);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.minProxyBalance = reader.int64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return { minProxyBalance: isSet(object.minProxyBalance) ? long_1.default.fromValue(object.minProxyBalance) : long_1.default.ZERO };
    },
    toJSON(message) {
        const obj = {};
        message.minProxyBalance !== undefined && (obj.minProxyBalance = (message.minProxyBalance || long_1.default.ZERO).toString());
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.minProxyBalance = (object.minProxyBalance !== undefined && object.minProxyBalance !== null)
            ? long_1.default.fromValue(object.minProxyBalance)
            : long_1.default.ZERO;
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
