// @axelar-network/axelarjs-types/build/commonjs/axelar/snapshot/v1beta1/service.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgServiceClientImpl = exports.protobufPackage = void 0;
/* eslint-disable */
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const tx_1 = require("./tx");
exports.protobufPackage = "axelar.snapshot.v1beta1";
class MsgServiceClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || "axelar.snapshot.v1beta1.MsgService";
        this.rpc = rpc;
        this.RegisterProxy = this.RegisterProxy.bind(this);
        this.DeactivateProxy = this.DeactivateProxy.bind(this);
    }
    RegisterProxy(request) {
        const data = tx_1.RegisterProxyRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RegisterProxy", data);
        return promise.then((data) => tx_1.RegisterProxyResponse.decode(new minimal_1.default.Reader(data)));
    }
    DeactivateProxy(request) {
        const data = tx_1.DeactivateProxyRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "DeactivateProxy", data);
        return promise.then((data) => tx_1.DeactivateProxyResponse.decode(new minimal_1.default.Reader(data)));
    }
}
exports.MsgServiceClientImpl = MsgServiceClientImpl;
