// @axelar-network/axelarjs-types/build/commonjs/axelar/reward/v1beta1/params.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "axelar.reward.v1beta1";
function createBaseParams() {
    return { externalChainVotingInflationRate: new Uint8Array(), keyMgmtRelativeInflationRate: new Uint8Array() };
}
exports.Params = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.externalChainVotingInflationRate.length !== 0) {
            writer.uint32(10).bytes(message.externalChainVotingInflationRate);
        }
        if (message.keyMgmtRelativeInflationRate.length !== 0) {
            writer.uint32(18).bytes(message.keyMgmtRelativeInflationRate);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.externalChainVotingInflationRate = reader.bytes();
                    break;
                case 2:
                    message.keyMgmtRelativeInflationRate = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            externalChainVotingInflationRate: isSet(object.externalChainVotingInflationRate)
                ? bytesFromBase64(object.externalChainVotingInflationRate)
                : new Uint8Array(),
            keyMgmtRelativeInflationRate: isSet(object.keyMgmtRelativeInflationRate)
                ? bytesFromBase64(object.keyMgmtRelativeInflationRate)
                : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.externalChainVotingInflationRate !== undefined &&
            (obj.externalChainVotingInflationRate = base64FromBytes(message.externalChainVotingInflationRate !== undefined
                ? message.externalChainVotingInflationRate
                : new Uint8Array()));
        message.keyMgmtRelativeInflationRate !== undefined &&
            (obj.keyMgmtRelativeInflationRate = base64FromBytes(message.keyMgmtRelativeInflationRate !== undefined ? message.keyMgmtRelativeInflationRate : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.externalChainVotingInflationRate = object.externalChainVotingInflationRate ?? new Uint8Array();
        message.keyMgmtRelativeInflationRate = object.keyMgmtRelativeInflationRate ?? new Uint8Array();
        return message;
    },
};
var tsProtoGlobalThis = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (tsProtoGlobalThis.Buffer) {
        return Uint8Array.from(tsProtoGlobalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = tsProtoGlobalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (tsProtoGlobalThis.Buffer) {
        return tsProtoGlobalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(String.fromCharCode(byte));
        });
        return tsProtoGlobalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
