// @axelar-network/axelarjs-types/build/module/axelar/tss/tofnd/v1beta1/common.js
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "axelar.tss.tofnd.v1beta1";
export var KeyPresenceResponse_Response;
(function (KeyPresenceResponse_Response) {
    KeyPresenceResponse_Response[KeyPresenceResponse_Response["RESPONSE_UNSPECIFIED"] = 0] = "RESPONSE_UNSPECIFIED";
    KeyPresenceResponse_Response[KeyPresenceResponse_Response["RESPONSE_PRESENT"] = 1] = "RESPONSE_PRESENT";
    KeyPresenceResponse_Response[KeyPresenceResponse_Response["RESPONSE_ABSENT"] = 2] = "RESPONSE_ABSENT";
    KeyPresenceResponse_Response[KeyPresenceResponse_Response["RESPONSE_FAIL"] = 3] = "RESPONSE_FAIL";
    KeyPresenceResponse_Response[KeyPresenceResponse_Response["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(KeyPresenceResponse_Response || (KeyPresenceResponse_Response = {}));
export function keyPresenceResponse_ResponseFromJSON(object) {
    switch (object) {
        case 0:
        case "RESPONSE_UNSPECIFIED":
            return KeyPresenceResponse_Response.RESPONSE_UNSPECIFIED;
        case 1:
        case "RESPONSE_PRESENT":
            return KeyPresenceResponse_Response.RESPONSE_PRESENT;
        case 2:
        case "RESPONSE_ABSENT":
            return KeyPresenceResponse_Response.RESPONSE_ABSENT;
        case 3:
        case "RESPONSE_FAIL":
            return KeyPresenceResponse_Response.RESPONSE_FAIL;
        case -1:
        case "UNRECOGNIZED":
        default:
            return KeyPresenceResponse_Response.UNRECOGNIZED;
    }
}
export function keyPresenceResponse_ResponseToJSON(object) {
    switch (object) {
        case KeyPresenceResponse_Response.RESPONSE_UNSPECIFIED:
            return "RESPONSE_UNSPECIFIED";
        case KeyPresenceResponse_Response.RESPONSE_PRESENT:
            return "RESPONSE_PRESENT";
        case KeyPresenceResponse_Response.RESPONSE_ABSENT:
            return "RESPONSE_ABSENT";
        case KeyPresenceResponse_Response.RESPONSE_FAIL:
            return "RESPONSE_FAIL";
        case KeyPresenceResponse_Response.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseKeyPresenceRequest() {
    return { keyUid: "", pubKey: new Uint8Array() };
}
export const KeyPresenceRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keyUid !== "") {
            writer.uint32(10).string(message.keyUid);
        }
        if (message.pubKey.length !== 0) {
            writer.uint32(18).bytes(message.pubKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseKeyPresenceRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.keyUid = reader.string();
                    break;
                case 2:
                    message.pubKey = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyUid: isSet(object.keyUid) ? String(object.keyUid) : "",
            pubKey: isSet(object.pubKey) ? bytesFromBase64(object.pubKey) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.keyUid !== undefined && (obj.keyUid = message.keyUid);
        message.pubKey !== undefined &&
            (obj.pubKey = base64FromBytes(message.pubKey !== undefined ? message.pubKey : new Uint8Array()));
        return obj;
    },
    create(base) {
        return KeyPresenceRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseKeyPresenceRequest();
        message.keyUid = object.keyUid ?? "";
        message.pubKey = object.pubKey ?? new Uint8Array();
        return message;
    },
};
function createBaseKeyPresenceResponse() {
    return { response: 0 };
}
export const KeyPresenceResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.response !== 0) {
            writer.uint32(8).int32(message.response);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseKeyPresenceResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.response = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return { response: isSet(object.response) ? keyPresenceResponse_ResponseFromJSON(object.response) : 0 };
    },
    toJSON(message) {
        const obj = {};
        message.response !== undefined && (obj.response = keyPresenceResponse_ResponseToJSON(message.response));
        return obj;
    },
    create(base) {
        return KeyPresenceResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseKeyPresenceResponse();
        message.response = object.response ?? 0;
        return message;
    },
};
var tsProtoGlobalThis = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (tsProtoGlobalThis.Buffer) {
        return Uint8Array.from(tsProtoGlobalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = tsProtoGlobalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (tsProtoGlobalThis.Buffer) {
        return tsProtoGlobalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(String.fromCharCode(byte));
        });
        return tsProtoGlobalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
