// @axelar-network/axelarjs-types/build/module/axelar/tss/v1beta1/tx.js
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { PollKey } from "../../vote/exported/v1beta1/types";
import { keyRoleFromJSON, keyRoleToJSON, SigKeyPair } from "../exported/v1beta1/types";
import { MessageOut_KeygenResult, MessageOut_SignResult, TrafficOut } from "../tofnd/v1beta1/tofnd";
import { KeyInfo } from "./types";
export const protobufPackage = "axelar.tss.v1beta1";
function createBaseStartKeygenRequest() {
    return { sender: "", keyInfo: undefined };
}
export const StartKeygenRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender !== "") {
            writer.uint32(10).string(message.sender);
        }
        if (message.keyInfo !== undefined) {
            KeyInfo.encode(message.keyInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStartKeygenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.string();
                    break;
                case 2:
                    message.keyInfo = KeyInfo.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? String(object.sender) : "",
            keyInfo: isSet(object.keyInfo) ? KeyInfo.fromJSON(object.keyInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined && (obj.sender = message.sender);
        message.keyInfo !== undefined && (obj.keyInfo = message.keyInfo ? KeyInfo.toJSON(message.keyInfo) : undefined);
        return obj;
    },
    create(base) {
        return StartKeygenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseStartKeygenRequest();
        message.sender = object.sender ?? "";
        message.keyInfo = (object.keyInfo !== undefined && object.keyInfo !== null)
            ? KeyInfo.fromPartial(object.keyInfo)
            : undefined;
        return message;
    },
};
function createBaseStartKeygenResponse() {
    return {};
}
export const StartKeygenResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseStartKeygenResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return StartKeygenResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseStartKeygenResponse();
        return message;
    },
};
function createBaseRotateKeyRequest() {
    return { sender: new Uint8Array(), chain: "", keyRole: 0, keyId: "" };
}
export const RotateKeyRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.keyRole !== 0) {
            writer.uint32(24).int32(message.keyRole);
        }
        if (message.keyId !== "") {
            writer.uint32(34).string(message.keyId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRotateKeyRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.keyRole = reader.int32();
                    break;
                case 4:
                    message.keyId = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            keyRole: isSet(object.keyRole) ? keyRoleFromJSON(object.keyRole) : 0,
            keyId: isSet(object.keyId) ? String(object.keyId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.keyRole !== undefined && (obj.keyRole = keyRoleToJSON(message.keyRole));
        message.keyId !== undefined && (obj.keyId = message.keyId);
        return obj;
    },
    create(base) {
        return RotateKeyRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRotateKeyRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.keyRole = object.keyRole ?? 0;
        message.keyId = object.keyId ?? "";
        return message;
    },
};
function createBaseRotateKeyResponse() {
    return {};
}
export const RotateKeyResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRotateKeyResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return RotateKeyResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseRotateKeyResponse();
        return message;
    },
};
function createBaseProcessKeygenTrafficRequest() {
    return { sender: new Uint8Array(), sessionId: "", payload: undefined };
}
export const ProcessKeygenTrafficRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.sessionId !== "") {
            writer.uint32(18).string(message.sessionId);
        }
        if (message.payload !== undefined) {
            TrafficOut.encode(message.payload, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProcessKeygenTrafficRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.sessionId = reader.string();
                    break;
                case 3:
                    message.payload = TrafficOut.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            sessionId: isSet(object.sessionId) ? String(object.sessionId) : "",
            payload: isSet(object.payload) ? TrafficOut.fromJSON(object.payload) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.sessionId !== undefined && (obj.sessionId = message.sessionId);
        message.payload !== undefined && (obj.payload = message.payload ? TrafficOut.toJSON(message.payload) : undefined);
        return obj;
    },
    create(base) {
        return ProcessKeygenTrafficRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProcessKeygenTrafficRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.sessionId = object.sessionId ?? "";
        message.payload = (object.payload !== undefined && object.payload !== null)
            ? TrafficOut.fromPartial(object.payload)
            : undefined;
        return message;
    },
};
function createBaseProcessKeygenTrafficResponse() {
    return {};
}
export const ProcessKeygenTrafficResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProcessKeygenTrafficResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return ProcessKeygenTrafficResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseProcessKeygenTrafficResponse();
        return message;
    },
};
function createBaseProcessSignTrafficRequest() {
    return { sender: new Uint8Array(), sessionId: "", payload: undefined };
}
export const ProcessSignTrafficRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.sessionId !== "") {
            writer.uint32(18).string(message.sessionId);
        }
        if (message.payload !== undefined) {
            TrafficOut.encode(message.payload, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProcessSignTrafficRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.sessionId = reader.string();
                    break;
                case 3:
                    message.payload = TrafficOut.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            sessionId: isSet(object.sessionId) ? String(object.sessionId) : "",
            payload: isSet(object.payload) ? TrafficOut.fromJSON(object.payload) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.sessionId !== undefined && (obj.sessionId = message.sessionId);
        message.payload !== undefined && (obj.payload = message.payload ? TrafficOut.toJSON(message.payload) : undefined);
        return obj;
    },
    create(base) {
        return ProcessSignTrafficRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProcessSignTrafficRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.sessionId = object.sessionId ?? "";
        message.payload = (object.payload !== undefined && object.payload !== null)
            ? TrafficOut.fromPartial(object.payload)
            : undefined;
        return message;
    },
};
function createBaseProcessSignTrafficResponse() {
    return {};
}
export const ProcessSignTrafficResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProcessSignTrafficResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return ProcessSignTrafficResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseProcessSignTrafficResponse();
        return message;
    },
};
function createBaseVotePubKeyRequest() {
    return { sender: new Uint8Array(), pollKey: undefined, result: undefined };
}
export const VotePubKeyRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.pollKey !== undefined) {
            PollKey.encode(message.pollKey, writer.uint32(18).fork()).ldelim();
        }
        if (message.result !== undefined) {
            MessageOut_KeygenResult.encode(message.result, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVotePubKeyRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.pollKey = PollKey.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.result = MessageOut_KeygenResult.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            pollKey: isSet(object.pollKey) ? PollKey.fromJSON(object.pollKey) : undefined,
            result: isSet(object.result) ? MessageOut_KeygenResult.fromJSON(object.result) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.pollKey !== undefined && (obj.pollKey = message.pollKey ? PollKey.toJSON(message.pollKey) : undefined);
        message.result !== undefined &&
            (obj.result = message.result ? MessageOut_KeygenResult.toJSON(message.result) : undefined);
        return obj;
    },
    create(base) {
        return VotePubKeyRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVotePubKeyRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.pollKey = (object.pollKey !== undefined && object.pollKey !== null)
            ? PollKey.fromPartial(object.pollKey)
            : undefined;
        message.result = (object.result !== undefined && object.result !== null)
            ? MessageOut_KeygenResult.fromPartial(object.result)
            : undefined;
        return message;
    },
};
function createBaseVotePubKeyResponse() {
    return { log: "" };
}
export const VotePubKeyResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.log !== "") {
            writer.uint32(10).string(message.log);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVotePubKeyResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.log = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return { log: isSet(object.log) ? String(object.log) : "" };
    },
    toJSON(message) {
        const obj = {};
        message.log !== undefined && (obj.log = message.log);
        return obj;
    },
    create(base) {
        return VotePubKeyResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVotePubKeyResponse();
        message.log = object.log ?? "";
        return message;
    },
};
function createBaseVoteSigRequest() {
    return { sender: new Uint8Array(), pollKey: undefined, result: undefined };
}
export const VoteSigRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.pollKey !== undefined) {
            PollKey.encode(message.pollKey, writer.uint32(18).fork()).ldelim();
        }
        if (message.result !== undefined) {
            MessageOut_SignResult.encode(message.result, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVoteSigRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.pollKey = PollKey.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.result = MessageOut_SignResult.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            pollKey: isSet(object.pollKey) ? PollKey.fromJSON(object.pollKey) : undefined,
            result: isSet(object.result) ? MessageOut_SignResult.fromJSON(object.result) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.pollKey !== undefined && (obj.pollKey = message.pollKey ? PollKey.toJSON(message.pollKey) : undefined);
        message.result !== undefined &&
            (obj.result = message.result ? MessageOut_SignResult.toJSON(message.result) : undefined);
        return obj;
    },
    create(base) {
        return VoteSigRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVoteSigRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.pollKey = (object.pollKey !== undefined && object.pollKey !== null)
            ? PollKey.fromPartial(object.pollKey)
            : undefined;
        message.result = (object.result !== undefined && object.result !== null)
            ? MessageOut_SignResult.fromPartial(object.result)
            : undefined;
        return message;
    },
};
function createBaseVoteSigResponse() {
    return { log: "" };
}
export const VoteSigResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.log !== "") {
            writer.uint32(10).string(message.log);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVoteSigResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.log = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return { log: isSet(object.log) ? String(object.log) : "" };
    },
    toJSON(message) {
        const obj = {};
        message.log !== undefined && (obj.log = message.log);
        return obj;
    },
    create(base) {
        return VoteSigResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVoteSigResponse();
        message.log = object.log ?? "";
        return message;
    },
};
function createBaseHeartBeatRequest() {
    return { sender: new Uint8Array(), keyIds: [] };
}
export const HeartBeatRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        for (const v of message.keyIds) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseHeartBeatRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.keyIds.push(reader.string());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            keyIds: Array.isArray(object?.keyIds) ? object.keyIds.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        if (message.keyIds) {
            obj.keyIds = message.keyIds.map((e) => e);
        }
        else {
            obj.keyIds = [];
        }
        return obj;
    },
    create(base) {
        return HeartBeatRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseHeartBeatRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.keyIds = object.keyIds?.map((e) => e) || [];
        return message;
    },
};
function createBaseHeartBeatResponse() {
    return {};
}
export const HeartBeatResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseHeartBeatResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return HeartBeatResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseHeartBeatResponse();
        return message;
    },
};
function createBaseRegisterExternalKeysRequest() {
    return { sender: new Uint8Array(), chain: "", externalKeys: [] };
}
export const RegisterExternalKeysRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        for (const v of message.externalKeys) {
            RegisterExternalKeysRequest_ExternalKey.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterExternalKeysRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.externalKeys.push(RegisterExternalKeysRequest_ExternalKey.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            externalKeys: Array.isArray(object?.externalKeys)
                ? object.externalKeys.map((e) => RegisterExternalKeysRequest_ExternalKey.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        if (message.externalKeys) {
            obj.externalKeys = message.externalKeys.map((e) => e ? RegisterExternalKeysRequest_ExternalKey.toJSON(e) : undefined);
        }
        else {
            obj.externalKeys = [];
        }
        return obj;
    },
    create(base) {
        return RegisterExternalKeysRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterExternalKeysRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.externalKeys = object.externalKeys?.map((e) => RegisterExternalKeysRequest_ExternalKey.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseRegisterExternalKeysRequest_ExternalKey() {
    return { id: "", pubKey: new Uint8Array() };
}
export const RegisterExternalKeysRequest_ExternalKey = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        if (message.pubKey.length !== 0) {
            writer.uint32(18).bytes(message.pubKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterExternalKeysRequest_ExternalKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.id = reader.string();
                    break;
                case 2:
                    message.pubKey = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? String(object.id) : "",
            pubKey: isSet(object.pubKey) ? bytesFromBase64(object.pubKey) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.id !== undefined && (obj.id = message.id);
        message.pubKey !== undefined &&
            (obj.pubKey = base64FromBytes(message.pubKey !== undefined ? message.pubKey : new Uint8Array()));
        return obj;
    },
    create(base) {
        return RegisterExternalKeysRequest_ExternalKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterExternalKeysRequest_ExternalKey();
        message.id = object.id ?? "";
        message.pubKey = object.pubKey ?? new Uint8Array();
        return message;
    },
};
function createBaseRegisterExternalKeysResponse() {
    return {};
}
export const RegisterExternalKeysResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterExternalKeysResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return RegisterExternalKeysResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseRegisterExternalKeysResponse();
        return message;
    },
};
function createBaseSubmitMultisigPubKeysRequest() {
    return { sender: new Uint8Array(), keyId: "", sigKeyPairs: [] };
}
export const SubmitMultisigPubKeysRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.keyId !== "") {
            writer.uint32(18).string(message.keyId);
        }
        for (const v of message.sigKeyPairs) {
            SigKeyPair.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubmitMultisigPubKeysRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.keyId = reader.string();
                    break;
                case 3:
                    message.sigKeyPairs.push(SigKeyPair.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            keyId: isSet(object.keyId) ? String(object.keyId) : "",
            sigKeyPairs: Array.isArray(object?.sigKeyPairs) ? object.sigKeyPairs.map((e) => SigKeyPair.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.keyId !== undefined && (obj.keyId = message.keyId);
        if (message.sigKeyPairs) {
            obj.sigKeyPairs = message.sigKeyPairs.map((e) => e ? SigKeyPair.toJSON(e) : undefined);
        }
        else {
            obj.sigKeyPairs = [];
        }
        return obj;
    },
    create(base) {
        return SubmitMultisigPubKeysRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubmitMultisigPubKeysRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.keyId = object.keyId ?? "";
        message.sigKeyPairs = object.sigKeyPairs?.map((e) => SigKeyPair.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSubmitMultisigPubKeysResponse() {
    return {};
}
export const SubmitMultisigPubKeysResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubmitMultisigPubKeysResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return SubmitMultisigPubKeysResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseSubmitMultisigPubKeysResponse();
        return message;
    },
};
function createBaseSubmitMultisigSignaturesRequest() {
    return { sender: new Uint8Array(), sigId: "", signatures: [] };
}
export const SubmitMultisigSignaturesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.sigId !== "") {
            writer.uint32(18).string(message.sigId);
        }
        for (const v of message.signatures) {
            writer.uint32(26).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubmitMultisigSignaturesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.sigId = reader.string();
                    break;
                case 3:
                    message.signatures.push(reader.bytes());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            sigId: isSet(object.sigId) ? String(object.sigId) : "",
            signatures: Array.isArray(object?.signatures) ? object.signatures.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.sigId !== undefined && (obj.sigId = message.sigId);
        if (message.signatures) {
            obj.signatures = message.signatures.map((e) => base64FromBytes(e !== undefined ? e : new Uint8Array()));
        }
        else {
            obj.signatures = [];
        }
        return obj;
    },
    create(base) {
        return SubmitMultisigSignaturesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubmitMultisigSignaturesRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.sigId = object.sigId ?? "";
        message.signatures = object.signatures?.map((e) => e) || [];
        return message;
    },
};
function createBaseSubmitMultisigSignaturesResponse() {
    return {};
}
export const SubmitMultisigSignaturesResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubmitMultisigSignaturesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return SubmitMultisigSignaturesResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseSubmitMultisigSignaturesResponse();
        return message;
    },
};
var tsProtoGlobalThis = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (tsProtoGlobalThis.Buffer) {
        return Uint8Array.from(tsProtoGlobalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = tsProtoGlobalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (tsProtoGlobalThis.Buffer) {
        return tsProtoGlobalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(String.fromCharCode(byte));
        });
        return tsProtoGlobalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
