// @axelar-network/axelarjs-types/build/module/axelar/snapshot/v1beta1/params.js
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "axelar.snapshot.v1beta1";
function createBaseParams() {
    return { minProxyBalance: Long.ZERO };
}
export const Params = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.minProxyBalance.isZero()) {
            writer.uint32(8).int64(message.minProxyBalance);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.minProxyBalance = reader.int64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return { minProxyBalance: isSet(object.minProxyBalance) ? Long.fromValue(object.minProxyBalance) : Long.ZERO };
    },
    toJSON(message) {
        const obj = {};
        message.minProxyBalance !== undefined && (obj.minProxyBalance = (message.minProxyBalance || Long.ZERO).toString());
        return obj;
    },
    create(base) {
        return Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.minProxyBalance = (object.minProxyBalance !== undefined && object.minProxyBalance !== null)
            ? Long.fromValue(object.minProxyBalance)
            : Long.ZERO;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
