// @axelar-network/axelarjs-types/build/module/axelar/snapshot/v1beta1/service.js
/* eslint-disable */
import _m0 from "protobufjs/minimal";
import { DeactivateProxyRequest, DeactivateProxyResponse, RegisterProxyRequest, RegisterProxyResponse } from "./tx";
export const protobufPackage = "axelar.snapshot.v1beta1";
export class MsgServiceClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || "axelar.snapshot.v1beta1.MsgService";
        this.rpc = rpc;
        this.RegisterProxy = this.RegisterProxy.bind(this);
        this.DeactivateProxy = this.DeactivateProxy.bind(this);
    }
    RegisterProxy(request) {
        const data = RegisterProxyRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RegisterProxy", data);
        return promise.then((data) => RegisterProxyResponse.decode(new _m0.Reader(data)));
    }
    DeactivateProxy(request) {
        const data = DeactivateProxyRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "DeactivateProxy", data);
        return promise.then((data) => DeactivateProxyResponse.decode(new _m0.Reader(data)));
    }
}
