// @axelar-network/axelarjs-types/build/module/axelar/utils/v1beta1/bitmap.js
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "axelar.utils.v1beta1";
function createBaseBitmap() {
    return { trueCountCache: undefined };
}
export const Bitmap = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.trueCountCache !== undefined) {
            CircularBuffer.encode(message.trueCountCache, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBitmap();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 2:
                    message.trueCountCache = CircularBuffer.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            trueCountCache: isSet(object.trueCountCache) ? CircularBuffer.fromJSON(object.trueCountCache) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.trueCountCache !== undefined &&
            (obj.trueCountCache = message.trueCountCache ? CircularBuffer.toJSON(message.trueCountCache) : undefined);
        return obj;
    },
    create(base) {
        return Bitmap.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBitmap();
        message.trueCountCache = (object.trueCountCache !== undefined && object.trueCountCache !== null)
            ? CircularBuffer.fromPartial(object.trueCountCache)
            : undefined;
        return message;
    },
};
function createBaseCircularBuffer() {
    return { cumulativeValue: [], index: 0, maxSize: 0 };
}
export const CircularBuffer = {
    encode(message, writer = _m0.Writer.create()) {
        writer.uint32(10).fork();
        for (const v of message.cumulativeValue) {
            writer.uint64(v);
        }
        writer.ldelim();
        if (message.index !== 0) {
            writer.uint32(16).int32(message.index);
        }
        if (message.maxSize !== 0) {
            writer.uint32(24).int32(message.maxSize);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCircularBuffer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if ((tag & 7) === 2) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.cumulativeValue.push(reader.uint64());
                        }
                    }
                    else {
                        message.cumulativeValue.push(reader.uint64());
                    }
                    break;
                case 2:
                    message.index = reader.int32();
                    break;
                case 3:
                    message.maxSize = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            cumulativeValue: Array.isArray(object?.cumulativeValue)
                ? object.cumulativeValue.map((e) => Long.fromValue(e))
                : [],
            index: isSet(object.index) ? Number(object.index) : 0,
            maxSize: isSet(object.maxSize) ? Number(object.maxSize) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.cumulativeValue) {
            obj.cumulativeValue = message.cumulativeValue.map((e) => (e || Long.UZERO).toString());
        }
        else {
            obj.cumulativeValue = [];
        }
        message.index !== undefined && (obj.index = Math.round(message.index));
        message.maxSize !== undefined && (obj.maxSize = Math.round(message.maxSize));
        return obj;
    },
    create(base) {
        return CircularBuffer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCircularBuffer();
        message.cumulativeValue = object.cumulativeValue?.map((e) => Long.fromValue(e)) || [];
        message.index = object.index ?? 0;
        message.maxSize = object.maxSize ?? 0;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
