// @axelar-network/axelarjs-types/build/commonjs/axelar/axelarnet/v1beta1/types.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fee = exports.Asset = exports.CosmosChain = exports.IBCTransfer = exports.iBCTransfer_StatusToJSON = exports.iBCTransfer_StatusFromJSON = exports.IBCTransfer_Status = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const coin_1 = require("../../../cosmos/base/v1beta1/coin");
exports.protobufPackage = "axelar.axelarnet.v1beta1";
var IBCTransfer_Status;
(function (IBCTransfer_Status) {
    IBCTransfer_Status[IBCTransfer_Status["STATUS_UNSPECIFIED"] = 0] = "STATUS_UNSPECIFIED";
    IBCTransfer_Status[IBCTransfer_Status["STATUS_PENDING"] = 1] = "STATUS_PENDING";
    IBCTransfer_Status[IBCTransfer_Status["STATUS_COMPLETED"] = 2] = "STATUS_COMPLETED";
    IBCTransfer_Status[IBCTransfer_Status["STATUS_FAILED"] = 3] = "STATUS_FAILED";
    IBCTransfer_Status[IBCTransfer_Status["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(IBCTransfer_Status = exports.IBCTransfer_Status || (exports.IBCTransfer_Status = {}));
function iBCTransfer_StatusFromJSON(object) {
    switch (object) {
        case 0:
        case "STATUS_UNSPECIFIED":
            return IBCTransfer_Status.STATUS_UNSPECIFIED;
        case 1:
        case "STATUS_PENDING":
            return IBCTransfer_Status.STATUS_PENDING;
        case 2:
        case "STATUS_COMPLETED":
            return IBCTransfer_Status.STATUS_COMPLETED;
        case 3:
        case "STATUS_FAILED":
            return IBCTransfer_Status.STATUS_FAILED;
        case -1:
        case "UNRECOGNIZED":
        default:
            return IBCTransfer_Status.UNRECOGNIZED;
    }
}
exports.iBCTransfer_StatusFromJSON = iBCTransfer_StatusFromJSON;
function iBCTransfer_StatusToJSON(object) {
    switch (object) {
        case IBCTransfer_Status.STATUS_UNSPECIFIED:
            return "STATUS_UNSPECIFIED";
        case IBCTransfer_Status.STATUS_PENDING:
            return "STATUS_PENDING";
        case IBCTransfer_Status.STATUS_COMPLETED:
            return "STATUS_COMPLETED";
        case IBCTransfer_Status.STATUS_FAILED:
            return "STATUS_FAILED";
        case IBCTransfer_Status.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.iBCTransfer_StatusToJSON = iBCTransfer_StatusToJSON;
function createBaseIBCTransfer() {
    return {
        sender: new Uint8Array(),
        receiver: "",
        token: undefined,
        portId: "",
        channelId: "",
        sequence: long_1.default.UZERO,
        id: long_1.default.UZERO,
        status: 0,
    };
}
exports.IBCTransfer = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.receiver !== "") {
            writer.uint32(18).string(message.receiver);
        }
        if (message.token !== undefined) {
            coin_1.Coin.encode(message.token, writer.uint32(26).fork()).ldelim();
        }
        if (message.portId !== "") {
            writer.uint32(34).string(message.portId);
        }
        if (message.channelId !== "") {
            writer.uint32(42).string(message.channelId);
        }
        if (!message.sequence.isZero()) {
            writer.uint32(48).uint64(message.sequence);
        }
        if (!message.id.isZero()) {
            writer.uint32(56).uint64(message.id);
        }
        if (message.status !== 0) {
            writer.uint32(64).int32(message.status);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseIBCTransfer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.receiver = reader.string();
                    break;
                case 3:
                    message.token = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.portId = reader.string();
                    break;
                case 5:
                    message.channelId = reader.string();
                    break;
                case 6:
                    message.sequence = reader.uint64();
                    break;
                case 7:
                    message.id = reader.uint64();
                    break;
                case 8:
                    message.status = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            receiver: isSet(object.receiver) ? String(object.receiver) : "",
            token: isSet(object.token) ? coin_1.Coin.fromJSON(object.token) : undefined,
            portId: isSet(object.portId) ? String(object.portId) : "",
            channelId: isSet(object.channelId) ? String(object.channelId) : "",
            sequence: isSet(object.sequence) ? long_1.default.fromValue(object.sequence) : long_1.default.UZERO,
            id: isSet(object.id) ? long_1.default.fromValue(object.id) : long_1.default.UZERO,
            status: isSet(object.status) ? iBCTransfer_StatusFromJSON(object.status) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.receiver !== undefined && (obj.receiver = message.receiver);
        message.token !== undefined && (obj.token = message.token ? coin_1.Coin.toJSON(message.token) : undefined);
        message.portId !== undefined && (obj.portId = message.portId);
        message.channelId !== undefined && (obj.channelId = message.channelId);
        message.sequence !== undefined && (obj.sequence = (message.sequence || long_1.default.UZERO).toString());
        message.id !== undefined && (obj.id = (message.id || long_1.default.UZERO).toString());
        message.status !== undefined && (obj.status = iBCTransfer_StatusToJSON(message.status));
        return obj;
    },
    create(base) {
        return exports.IBCTransfer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseIBCTransfer();
        message.sender = object.sender ?? new Uint8Array();
        message.receiver = object.receiver ?? "";
        message.token = (object.token !== undefined && object.token !== null) ? coin_1.Coin.fromPartial(object.token) : undefined;
        message.portId = object.portId ?? "";
        message.channelId = object.channelId ?? "";
        message.sequence = (object.sequence !== undefined && object.sequence !== null)
            ? long_1.default.fromValue(object.sequence)
            : long_1.default.UZERO;
        message.id = (object.id !== undefined && object.id !== null) ? long_1.default.fromValue(object.id) : long_1.default.UZERO;
        message.status = object.status ?? 0;
        return message;
    },
};
function createBaseCosmosChain() {
    return { name: "", ibcPath: "", assets: [], addrPrefix: "" };
}
exports.CosmosChain = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.name !== "") {
            writer.uint32(10).string(message.name);
        }
        if (message.ibcPath !== "") {
            writer.uint32(18).string(message.ibcPath);
        }
        for (const v of message.assets) {
            exports.Asset.encode(v, writer.uint32(26).fork()).ldelim();
        }
        if (message.addrPrefix !== "") {
            writer.uint32(34).string(message.addrPrefix);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCosmosChain();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.name = reader.string();
                    break;
                case 2:
                    message.ibcPath = reader.string();
                    break;
                case 3:
                    message.assets.push(exports.Asset.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.addrPrefix = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            name: isSet(object.name) ? String(object.name) : "",
            ibcPath: isSet(object.ibcPath) ? String(object.ibcPath) : "",
            assets: Array.isArray(object?.assets) ? object.assets.map((e) => exports.Asset.fromJSON(e)) : [],
            addrPrefix: isSet(object.addrPrefix) ? String(object.addrPrefix) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.name !== undefined && (obj.name = message.name);
        message.ibcPath !== undefined && (obj.ibcPath = message.ibcPath);
        if (message.assets) {
            obj.assets = message.assets.map((e) => e ? exports.Asset.toJSON(e) : undefined);
        }
        else {
            obj.assets = [];
        }
        message.addrPrefix !== undefined && (obj.addrPrefix = message.addrPrefix);
        return obj;
    },
    create(base) {
        return exports.CosmosChain.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCosmosChain();
        message.name = object.name ?? "";
        message.ibcPath = object.ibcPath ?? "";
        message.assets = object.assets?.map((e) => exports.Asset.fromPartial(e)) || [];
        message.addrPrefix = object.addrPrefix ?? "";
        return message;
    },
};
function createBaseAsset() {
    return { denom: "", minAmount: new Uint8Array() };
}
exports.Asset = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.denom !== "") {
            writer.uint32(10).string(message.denom);
        }
        if (message.minAmount.length !== 0) {
            writer.uint32(18).bytes(message.minAmount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAsset();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.denom = reader.string();
                    break;
                case 2:
                    message.minAmount = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            denom: isSet(object.denom) ? String(object.denom) : "",
            minAmount: isSet(object.minAmount) ? bytesFromBase64(object.minAmount) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.denom !== undefined && (obj.denom = message.denom);
        message.minAmount !== undefined &&
            (obj.minAmount = base64FromBytes(message.minAmount !== undefined ? message.minAmount : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.Asset.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAsset();
        message.denom = object.denom ?? "";
        message.minAmount = object.minAmount ?? new Uint8Array();
        return message;
    },
};
function createBaseFee() {
    return { amount: undefined, recipient: new Uint8Array() };
}
exports.Fee = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.amount !== undefined) {
            coin_1.Coin.encode(message.amount, writer.uint32(10).fork()).ldelim();
        }
        if (message.recipient.length !== 0) {
            writer.uint32(18).bytes(message.recipient);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFee();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.amount = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.recipient = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            amount: isSet(object.amount) ? coin_1.Coin.fromJSON(object.amount) : undefined,
            recipient: isSet(object.recipient) ? bytesFromBase64(object.recipient) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.amount !== undefined && (obj.amount = message.amount ? coin_1.Coin.toJSON(message.amount) : undefined);
        message.recipient !== undefined &&
            (obj.recipient = base64FromBytes(message.recipient !== undefined ? message.recipient : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.Fee.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFee();
        message.amount = (object.amount !== undefined && object.amount !== null)
            ? coin_1.Coin.fromPartial(object.amount)
            : undefined;
        message.recipient = object.recipient ?? new Uint8Array();
        return message;
    },
};
var tsProtoGlobalThis = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (tsProtoGlobalThis.Buffer) {
        return Uint8Array.from(tsProtoGlobalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = tsProtoGlobalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (tsProtoGlobalThis.Buffer) {
        return tsProtoGlobalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(String.fromCharCode(byte));
        });
        return tsProtoGlobalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
