// @axelar-network/axelarjs-types/build/commonjs/axelar/nexus/v1beta1/params.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const threshold_1 = require("../../utils/v1beta1/threshold");
exports.protobufPackage = "axelar.nexus.v1beta1";
function createBaseParams() {
    return {
        chainActivationThreshold: undefined,
        chainMaintainerMissingVoteThreshold: undefined,
        chainMaintainerIncorrectVoteThreshold: undefined,
        chainMaintainerCheckWindow: 0,
    };
}
exports.Params = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.chainActivationThreshold !== undefined) {
            threshold_1.Threshold.encode(message.chainActivationThreshold, writer.uint32(10).fork()).ldelim();
        }
        if (message.chainMaintainerMissingVoteThreshold !== undefined) {
            threshold_1.Threshold.encode(message.chainMaintainerMissingVoteThreshold, writer.uint32(18).fork()).ldelim();
        }
        if (message.chainMaintainerIncorrectVoteThreshold !== undefined) {
            threshold_1.Threshold.encode(message.chainMaintainerIncorrectVoteThreshold, writer.uint32(26).fork()).ldelim();
        }
        if (message.chainMaintainerCheckWindow !== 0) {
            writer.uint32(32).int32(message.chainMaintainerCheckWindow);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.chainActivationThreshold = threshold_1.Threshold.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.chainMaintainerMissingVoteThreshold = threshold_1.Threshold.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.chainMaintainerIncorrectVoteThreshold = threshold_1.Threshold.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.chainMaintainerCheckWindow = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            chainActivationThreshold: isSet(object.chainActivationThreshold)
                ? threshold_1.Threshold.fromJSON(object.chainActivationThreshold)
                : undefined,
            chainMaintainerMissingVoteThreshold: isSet(object.chainMaintainerMissingVoteThreshold)
                ? threshold_1.Threshold.fromJSON(object.chainMaintainerMissingVoteThreshold)
                : undefined,
            chainMaintainerIncorrectVoteThreshold: isSet(object.chainMaintainerIncorrectVoteThreshold)
                ? threshold_1.Threshold.fromJSON(object.chainMaintainerIncorrectVoteThreshold)
                : undefined,
            chainMaintainerCheckWindow: isSet(object.chainMaintainerCheckWindow)
                ? Number(object.chainMaintainerCheckWindow)
                : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        message.chainActivationThreshold !== undefined && (obj.chainActivationThreshold = message.chainActivationThreshold
            ? threshold_1.Threshold.toJSON(message.chainActivationThreshold)
            : undefined);
        message.chainMaintainerMissingVoteThreshold !== undefined &&
            (obj.chainMaintainerMissingVoteThreshold = message.chainMaintainerMissingVoteThreshold
                ? threshold_1.Threshold.toJSON(message.chainMaintainerMissingVoteThreshold)
                : undefined);
        message.chainMaintainerIncorrectVoteThreshold !== undefined &&
            (obj.chainMaintainerIncorrectVoteThreshold = message.chainMaintainerIncorrectVoteThreshold
                ? threshold_1.Threshold.toJSON(message.chainMaintainerIncorrectVoteThreshold)
                : undefined);
        message.chainMaintainerCheckWindow !== undefined &&
            (obj.chainMaintainerCheckWindow = Math.round(message.chainMaintainerCheckWindow));
        return obj;
    },
    create(base) {
        return exports.Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.chainActivationThreshold =
            (object.chainActivationThreshold !== undefined && object.chainActivationThreshold !== null)
                ? threshold_1.Threshold.fromPartial(object.chainActivationThreshold)
                : undefined;
        message.chainMaintainerMissingVoteThreshold =
            (object.chainMaintainerMissingVoteThreshold !== undefined && object.chainMaintainerMissingVoteThreshold !== null)
                ? threshold_1.Threshold.fromPartial(object.chainMaintainerMissingVoteThreshold)
                : undefined;
        message.chainMaintainerIncorrectVoteThreshold =
            (object.chainMaintainerIncorrectVoteThreshold !== undefined &&
                object.chainMaintainerIncorrectVoteThreshold !== null)
                ? threshold_1.Threshold.fromPartial(object.chainMaintainerIncorrectVoteThreshold)
                : undefined;
        message.chainMaintainerCheckWindow = object.chainMaintainerCheckWindow ?? 0;
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
