// @axelar-network/axelarjs-types/build/commonjs/axelar/nexus/v1beta1/tx.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SetTransferRateLimitResponse = exports.SetTransferRateLimitRequest = exports.RegisterAssetFeeResponse = exports.RegisterAssetFeeRequest = exports.DeactivateChainResponse = exports.DeactivateChainRequest = exports.ActivateChainResponse = exports.ActivateChainRequest = exports.DeregisterChainMaintainerResponse = exports.DeregisterChainMaintainerRequest = exports.RegisterChainMaintainerResponse = exports.RegisterChainMaintainerRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const coin_1 = require("../../../cosmos/base/v1beta1/coin");
const duration_1 = require("../../../google/protobuf/duration");
const types_1 = require("../exported/v1beta1/types");
exports.protobufPackage = "axelar.nexus.v1beta1";
function createBaseRegisterChainMaintainerRequest() {
    return { sender: new Uint8Array(), chains: [] };
}
exports.RegisterChainMaintainerRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        for (const v of message.chains) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterChainMaintainerRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chains.push(reader.string());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chains: Array.isArray(object?.chains) ? object.chains.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        if (message.chains) {
            obj.chains = message.chains.map((e) => e);
        }
        else {
            obj.chains = [];
        }
        return obj;
    },
    create(base) {
        return exports.RegisterChainMaintainerRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterChainMaintainerRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chains = object.chains?.map((e) => e) || [];
        return message;
    },
};
function createBaseRegisterChainMaintainerResponse() {
    return {};
}
exports.RegisterChainMaintainerResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterChainMaintainerResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.RegisterChainMaintainerResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseRegisterChainMaintainerResponse();
        return message;
    },
};
function createBaseDeregisterChainMaintainerRequest() {
    return { sender: new Uint8Array(), chains: [] };
}
exports.DeregisterChainMaintainerRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        for (const v of message.chains) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeregisterChainMaintainerRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chains.push(reader.string());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chains: Array.isArray(object?.chains) ? object.chains.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        if (message.chains) {
            obj.chains = message.chains.map((e) => e);
        }
        else {
            obj.chains = [];
        }
        return obj;
    },
    create(base) {
        return exports.DeregisterChainMaintainerRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeregisterChainMaintainerRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chains = object.chains?.map((e) => e) || [];
        return message;
    },
};
function createBaseDeregisterChainMaintainerResponse() {
    return {};
}
exports.DeregisterChainMaintainerResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeregisterChainMaintainerResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.DeregisterChainMaintainerResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseDeregisterChainMaintainerResponse();
        return message;
    },
};
function createBaseActivateChainRequest() {
    return { sender: new Uint8Array(), chains: [] };
}
exports.ActivateChainRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        for (const v of message.chains) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseActivateChainRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chains.push(reader.string());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chains: Array.isArray(object?.chains) ? object.chains.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        if (message.chains) {
            obj.chains = message.chains.map((e) => e);
        }
        else {
            obj.chains = [];
        }
        return obj;
    },
    create(base) {
        return exports.ActivateChainRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseActivateChainRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chains = object.chains?.map((e) => e) || [];
        return message;
    },
};
function createBaseActivateChainResponse() {
    return {};
}
exports.ActivateChainResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseActivateChainResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.ActivateChainResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseActivateChainResponse();
        return message;
    },
};
function createBaseDeactivateChainRequest() {
    return { sender: new Uint8Array(), chains: [] };
}
exports.DeactivateChainRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        for (const v of message.chains) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeactivateChainRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chains.push(reader.string());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chains: Array.isArray(object?.chains) ? object.chains.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        if (message.chains) {
            obj.chains = message.chains.map((e) => e);
        }
        else {
            obj.chains = [];
        }
        return obj;
    },
    create(base) {
        return exports.DeactivateChainRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeactivateChainRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chains = object.chains?.map((e) => e) || [];
        return message;
    },
};
function createBaseDeactivateChainResponse() {
    return {};
}
exports.DeactivateChainResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeactivateChainResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.DeactivateChainResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseDeactivateChainResponse();
        return message;
    },
};
function createBaseRegisterAssetFeeRequest() {
    return { sender: new Uint8Array(), feeInfo: undefined };
}
exports.RegisterAssetFeeRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.feeInfo !== undefined) {
            types_1.FeeInfo.encode(message.feeInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterAssetFeeRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.feeInfo = types_1.FeeInfo.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            feeInfo: isSet(object.feeInfo) ? types_1.FeeInfo.fromJSON(object.feeInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.feeInfo !== undefined && (obj.feeInfo = message.feeInfo ? types_1.FeeInfo.toJSON(message.feeInfo) : undefined);
        return obj;
    },
    create(base) {
        return exports.RegisterAssetFeeRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterAssetFeeRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.feeInfo = (object.feeInfo !== undefined && object.feeInfo !== null)
            ? types_1.FeeInfo.fromPartial(object.feeInfo)
            : undefined;
        return message;
    },
};
function createBaseRegisterAssetFeeResponse() {
    return {};
}
exports.RegisterAssetFeeResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterAssetFeeResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.RegisterAssetFeeResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseRegisterAssetFeeResponse();
        return message;
    },
};
function createBaseSetTransferRateLimitRequest() {
    return { sender: new Uint8Array(), chain: "", limit: undefined, window: undefined };
}
exports.SetTransferRateLimitRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.limit !== undefined) {
            coin_1.Coin.encode(message.limit, writer.uint32(26).fork()).ldelim();
        }
        if (message.window !== undefined) {
            duration_1.Duration.encode(message.window, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetTransferRateLimitRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.limit = coin_1.Coin.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.window = duration_1.Duration.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            limit: isSet(object.limit) ? coin_1.Coin.fromJSON(object.limit) : undefined,
            window: isSet(object.window) ? duration_1.Duration.fromJSON(object.window) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.limit !== undefined && (obj.limit = message.limit ? coin_1.Coin.toJSON(message.limit) : undefined);
        message.window !== undefined && (obj.window = message.window ? duration_1.Duration.toJSON(message.window) : undefined);
        return obj;
    },
    create(base) {
        return exports.SetTransferRateLimitRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSetTransferRateLimitRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.limit = (object.limit !== undefined && object.limit !== null) ? coin_1.Coin.fromPartial(object.limit) : undefined;
        message.window = (object.window !== undefined && object.window !== null)
            ? duration_1.Duration.fromPartial(object.window)
            : undefined;
        return message;
    },
};
function createBaseSetTransferRateLimitResponse() {
    return {};
}
exports.SetTransferRateLimitResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetTransferRateLimitResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.SetTransferRateLimitResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseSetTransferRateLimitResponse();
        return message;
    },
};
var tsProtoGlobalThis = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (tsProtoGlobalThis.Buffer) {
        return Uint8Array.from(tsProtoGlobalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = tsProtoGlobalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (tsProtoGlobalThis.Buffer) {
        return tsProtoGlobalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(String.fromCharCode(byte));
        });
        return tsProtoGlobalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
