// @axelar-network/axelarjs-types/build/commonjs/axelar/evm/v1beta1/genesis.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState_Chain = exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const queuer_1 = require("../../utils/v1beta1/queuer");
const params_1 = require("./params");
const types_1 = require("./types");
exports.protobufPackage = "axelar.evm.v1beta1";
function createBaseGenesisState() {
    return { chains: [] };
}
exports.GenesisState = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.chains) {
            exports.GenesisState_Chain.encode(v, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 3:
                    message.chains.push(exports.GenesisState_Chain.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            chains: Array.isArray(object?.chains) ? object.chains.map((e) => exports.GenesisState_Chain.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.chains) {
            obj.chains = message.chains.map((e) => e ? exports.GenesisState_Chain.toJSON(e) : undefined);
        }
        else {
            obj.chains = [];
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState();
        message.chains = object.chains?.map((e) => exports.GenesisState_Chain.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGenesisState_Chain() {
    return {
        params: undefined,
        burnerInfos: [],
        commandQueue: undefined,
        confirmedDeposits: [],
        burnedDeposits: [],
        commandBatches: [],
        gateway: undefined,
        tokens: [],
        events: [],
        confirmedEventQueue: undefined,
        legacyConfirmedDeposits: [],
        legacyBurnedDeposits: [],
    };
}
exports.GenesisState_Chain = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.params !== undefined) {
            params_1.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.burnerInfos) {
            types_1.BurnerInfo.encode(v, writer.uint32(18).fork()).ldelim();
        }
        if (message.commandQueue !== undefined) {
            queuer_1.QueueState.encode(message.commandQueue, writer.uint32(26).fork()).ldelim();
        }
        for (const v of message.confirmedDeposits) {
            types_1.ERC20Deposit.encode(v, writer.uint32(34).fork()).ldelim();
        }
        for (const v of message.burnedDeposits) {
            types_1.ERC20Deposit.encode(v, writer.uint32(42).fork()).ldelim();
        }
        for (const v of message.commandBatches) {
            types_1.CommandBatchMetadata.encode(v, writer.uint32(66).fork()).ldelim();
        }
        if (message.gateway !== undefined) {
            types_1.Gateway.encode(message.gateway, writer.uint32(74).fork()).ldelim();
        }
        for (const v of message.tokens) {
            types_1.ERC20TokenMetadata.encode(v, writer.uint32(82).fork()).ldelim();
        }
        for (const v of message.events) {
            types_1.Event.encode(v, writer.uint32(90).fork()).ldelim();
        }
        if (message.confirmedEventQueue !== undefined) {
            queuer_1.QueueState.encode(message.confirmedEventQueue, writer.uint32(98).fork()).ldelim();
        }
        for (const v of message.legacyConfirmedDeposits) {
            types_1.ERC20Deposit.encode(v, writer.uint32(106).fork()).ldelim();
        }
        for (const v of message.legacyBurnedDeposits) {
            types_1.ERC20Deposit.encode(v, writer.uint32(114).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGenesisState_Chain();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.params = params_1.Params.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.burnerInfos.push(types_1.BurnerInfo.decode(reader, reader.uint32()));
                    break;
                case 3:
                    message.commandQueue = queuer_1.QueueState.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.confirmedDeposits.push(types_1.ERC20Deposit.decode(reader, reader.uint32()));
                    break;
                case 5:
                    message.burnedDeposits.push(types_1.ERC20Deposit.decode(reader, reader.uint32()));
                    break;
                case 8:
                    message.commandBatches.push(types_1.CommandBatchMetadata.decode(reader, reader.uint32()));
                    break;
                case 9:
                    message.gateway = types_1.Gateway.decode(reader, reader.uint32());
                    break;
                case 10:
                    message.tokens.push(types_1.ERC20TokenMetadata.decode(reader, reader.uint32()));
                    break;
                case 11:
                    message.events.push(types_1.Event.decode(reader, reader.uint32()));
                    break;
                case 12:
                    message.confirmedEventQueue = queuer_1.QueueState.decode(reader, reader.uint32());
                    break;
                case 13:
                    message.legacyConfirmedDeposits.push(types_1.ERC20Deposit.decode(reader, reader.uint32()));
                    break;
                case 14:
                    message.legacyBurnedDeposits.push(types_1.ERC20Deposit.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            params: isSet(object.params) ? params_1.Params.fromJSON(object.params) : undefined,
            burnerInfos: Array.isArray(object?.burnerInfos) ? object.burnerInfos.map((e) => types_1.BurnerInfo.fromJSON(e)) : [],
            commandQueue: isSet(object.commandQueue) ? queuer_1.QueueState.fromJSON(object.commandQueue) : undefined,
            confirmedDeposits: Array.isArray(object?.confirmedDeposits)
                ? object.confirmedDeposits.map((e) => types_1.ERC20Deposit.fromJSON(e))
                : [],
            burnedDeposits: Array.isArray(object?.burnedDeposits)
                ? object.burnedDeposits.map((e) => types_1.ERC20Deposit.fromJSON(e))
                : [],
            commandBatches: Array.isArray(object?.commandBatches)
                ? object.commandBatches.map((e) => types_1.CommandBatchMetadata.fromJSON(e))
                : [],
            gateway: isSet(object.gateway) ? types_1.Gateway.fromJSON(object.gateway) : undefined,
            tokens: Array.isArray(object?.tokens) ? object.tokens.map((e) => types_1.ERC20TokenMetadata.fromJSON(e)) : [],
            events: Array.isArray(object?.events) ? object.events.map((e) => types_1.Event.fromJSON(e)) : [],
            confirmedEventQueue: isSet(object.confirmedEventQueue)
                ? queuer_1.QueueState.fromJSON(object.confirmedEventQueue)
                : undefined,
            legacyConfirmedDeposits: Array.isArray(object?.legacyConfirmedDeposits)
                ? object.legacyConfirmedDeposits.map((e) => types_1.ERC20Deposit.fromJSON(e))
                : [],
            legacyBurnedDeposits: Array.isArray(object?.legacyBurnedDeposits)
                ? object.legacyBurnedDeposits.map((e) => types_1.ERC20Deposit.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.params !== undefined && (obj.params = message.params ? params_1.Params.toJSON(message.params) : undefined);
        if (message.burnerInfos) {
            obj.burnerInfos = message.burnerInfos.map((e) => e ? types_1.BurnerInfo.toJSON(e) : undefined);
        }
        else {
            obj.burnerInfos = [];
        }
        message.commandQueue !== undefined &&
            (obj.commandQueue = message.commandQueue ? queuer_1.QueueState.toJSON(message.commandQueue) : undefined);
        if (message.confirmedDeposits) {
            obj.confirmedDeposits = message.confirmedDeposits.map((e) => e ? types_1.ERC20Deposit.toJSON(e) : undefined);
        }
        else {
            obj.confirmedDeposits = [];
        }
        if (message.burnedDeposits) {
            obj.burnedDeposits = message.burnedDeposits.map((e) => e ? types_1.ERC20Deposit.toJSON(e) : undefined);
        }
        else {
            obj.burnedDeposits = [];
        }
        if (message.commandBatches) {
            obj.commandBatches = message.commandBatches.map((e) => e ? types_1.CommandBatchMetadata.toJSON(e) : undefined);
        }
        else {
            obj.commandBatches = [];
        }
        message.gateway !== undefined && (obj.gateway = message.gateway ? types_1.Gateway.toJSON(message.gateway) : undefined);
        if (message.tokens) {
            obj.tokens = message.tokens.map((e) => e ? types_1.ERC20TokenMetadata.toJSON(e) : undefined);
        }
        else {
            obj.tokens = [];
        }
        if (message.events) {
            obj.events = message.events.map((e) => e ? types_1.Event.toJSON(e) : undefined);
        }
        else {
            obj.events = [];
        }
        message.confirmedEventQueue !== undefined && (obj.confirmedEventQueue = message.confirmedEventQueue
            ? queuer_1.QueueState.toJSON(message.confirmedEventQueue)
            : undefined);
        if (message.legacyConfirmedDeposits) {
            obj.legacyConfirmedDeposits = message.legacyConfirmedDeposits.map((e) => e ? types_1.ERC20Deposit.toJSON(e) : undefined);
        }
        else {
            obj.legacyConfirmedDeposits = [];
        }
        if (message.legacyBurnedDeposits) {
            obj.legacyBurnedDeposits = message.legacyBurnedDeposits.map((e) => e ? types_1.ERC20Deposit.toJSON(e) : undefined);
        }
        else {
            obj.legacyBurnedDeposits = [];
        }
        return obj;
    },
    create(base) {
        return exports.GenesisState_Chain.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGenesisState_Chain();
        message.params = (object.params !== undefined && object.params !== null)
            ? params_1.Params.fromPartial(object.params)
            : undefined;
        message.burnerInfos = object.burnerInfos?.map((e) => types_1.BurnerInfo.fromPartial(e)) || [];
        message.commandQueue = (object.commandQueue !== undefined && object.commandQueue !== null)
            ? queuer_1.QueueState.fromPartial(object.commandQueue)
            : undefined;
        message.confirmedDeposits = object.confirmedDeposits?.map((e) => types_1.ERC20Deposit.fromPartial(e)) || [];
        message.burnedDeposits = object.burnedDeposits?.map((e) => types_1.ERC20Deposit.fromPartial(e)) || [];
        message.commandBatches = object.commandBatches?.map((e) => types_1.CommandBatchMetadata.fromPartial(e)) || [];
        message.gateway = (object.gateway !== undefined && object.gateway !== null)
            ? types_1.Gateway.fromPartial(object.gateway)
            : undefined;
        message.tokens = object.tokens?.map((e) => types_1.ERC20TokenMetadata.fromPartial(e)) || [];
        message.events = object.events?.map((e) => types_1.Event.fromPartial(e)) || [];
        message.confirmedEventQueue = (object.confirmedEventQueue !== undefined && object.confirmedEventQueue !== null)
            ? queuer_1.QueueState.fromPartial(object.confirmedEventQueue)
            : undefined;
        message.legacyConfirmedDeposits = object.legacyConfirmedDeposits?.map((e) => types_1.ERC20Deposit.fromPartial(e)) || [];
        message.legacyBurnedDeposits = object.legacyBurnedDeposits?.map((e) => types_1.ERC20Deposit.fromPartial(e)) || [];
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
