// @axelar-network/axelarjs-types/build/commonjs/axelar/evm/v1beta1/tx.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetryFailedEventResponse = exports.RetryFailedEventRequest = exports.AddChainResponse = exports.AddChainRequest = exports.SignCommandsResponse = exports.SignCommandsRequest = exports.CreateTransferOperatorshipResponse = exports.CreateTransferOperatorshipRequest = exports.CreateTransferOwnershipResponse = exports.CreateTransferOwnershipRequest = exports.CreatePendingTransfersResponse = exports.CreatePendingTransfersRequest = exports.CreateDeployTokenResponse = exports.CreateDeployTokenRequest = exports.CreateBurnTokensResponse = exports.CreateBurnTokensRequest = exports.LinkResponse = exports.LinkRequest = exports.ConfirmTransferKeyResponse = exports.ConfirmTransferKeyRequest = exports.ConfirmTokenResponse = exports.ConfirmTokenRequest = exports.ConfirmDepositResponse = exports.ConfirmDepositRequest = exports.ConfirmGatewayTxsResponse = exports.ConfirmGatewayTxsRequest = exports.ConfirmGatewayTxResponse = exports.ConfirmGatewayTxRequest = exports.SetGatewayResponse = exports.SetGatewayRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const types_1 = require("../../tss/exported/v1beta1/types");
const types_2 = require("./types");
exports.protobufPackage = "axelar.evm.v1beta1";
function createBaseSetGatewayRequest() {
    return { sender: new Uint8Array(), chain: "", address: new Uint8Array() };
}
exports.SetGatewayRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.address.length !== 0) {
            writer.uint32(26).bytes(message.address);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetGatewayRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.address = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            address: isSet(object.address) ? bytesFromBase64(object.address) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.address !== undefined &&
            (obj.address = base64FromBytes(message.address !== undefined ? message.address : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.SetGatewayRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSetGatewayRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.address = object.address ?? new Uint8Array();
        return message;
    },
};
function createBaseSetGatewayResponse() {
    return {};
}
exports.SetGatewayResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetGatewayResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.SetGatewayResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseSetGatewayResponse();
        return message;
    },
};
function createBaseConfirmGatewayTxRequest() {
    return { sender: new Uint8Array(), chain: "", txId: new Uint8Array() };
}
exports.ConfirmGatewayTxRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.txId.length !== 0) {
            writer.uint32(26).bytes(message.txId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmGatewayTxRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.txId = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            txId: isSet(object.txId) ? bytesFromBase64(object.txId) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.txId !== undefined &&
            (obj.txId = base64FromBytes(message.txId !== undefined ? message.txId : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.ConfirmGatewayTxRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConfirmGatewayTxRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.txId = object.txId ?? new Uint8Array();
        return message;
    },
};
function createBaseConfirmGatewayTxResponse() {
    return {};
}
exports.ConfirmGatewayTxResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmGatewayTxResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.ConfirmGatewayTxResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseConfirmGatewayTxResponse();
        return message;
    },
};
function createBaseConfirmGatewayTxsRequest() {
    return { sender: new Uint8Array(), chain: "", txIds: [] };
}
exports.ConfirmGatewayTxsRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        for (const v of message.txIds) {
            writer.uint32(26).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmGatewayTxsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.txIds.push(reader.bytes());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            txIds: Array.isArray(object?.txIds) ? object.txIds.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        if (message.txIds) {
            obj.txIds = message.txIds.map((e) => base64FromBytes(e !== undefined ? e : new Uint8Array()));
        }
        else {
            obj.txIds = [];
        }
        return obj;
    },
    create(base) {
        return exports.ConfirmGatewayTxsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConfirmGatewayTxsRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.txIds = object.txIds?.map((e) => e) || [];
        return message;
    },
};
function createBaseConfirmGatewayTxsResponse() {
    return {};
}
exports.ConfirmGatewayTxsResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmGatewayTxsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.ConfirmGatewayTxsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseConfirmGatewayTxsResponse();
        return message;
    },
};
function createBaseConfirmDepositRequest() {
    return {
        sender: new Uint8Array(),
        chain: "",
        txId: new Uint8Array(),
        amount: new Uint8Array(),
        burnerAddress: new Uint8Array(),
    };
}
exports.ConfirmDepositRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.txId.length !== 0) {
            writer.uint32(26).bytes(message.txId);
        }
        if (message.amount.length !== 0) {
            writer.uint32(34).bytes(message.amount);
        }
        if (message.burnerAddress.length !== 0) {
            writer.uint32(42).bytes(message.burnerAddress);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmDepositRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.txId = reader.bytes();
                    break;
                case 4:
                    message.amount = reader.bytes();
                    break;
                case 5:
                    message.burnerAddress = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            txId: isSet(object.txId) ? bytesFromBase64(object.txId) : new Uint8Array(),
            amount: isSet(object.amount) ? bytesFromBase64(object.amount) : new Uint8Array(),
            burnerAddress: isSet(object.burnerAddress) ? bytesFromBase64(object.burnerAddress) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.txId !== undefined &&
            (obj.txId = base64FromBytes(message.txId !== undefined ? message.txId : new Uint8Array()));
        message.amount !== undefined &&
            (obj.amount = base64FromBytes(message.amount !== undefined ? message.amount : new Uint8Array()));
        message.burnerAddress !== undefined &&
            (obj.burnerAddress = base64FromBytes(message.burnerAddress !== undefined ? message.burnerAddress : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.ConfirmDepositRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConfirmDepositRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.txId = object.txId ?? new Uint8Array();
        message.amount = object.amount ?? new Uint8Array();
        message.burnerAddress = object.burnerAddress ?? new Uint8Array();
        return message;
    },
};
function createBaseConfirmDepositResponse() {
    return {};
}
exports.ConfirmDepositResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmDepositResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.ConfirmDepositResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseConfirmDepositResponse();
        return message;
    },
};
function createBaseConfirmTokenRequest() {
    return { sender: new Uint8Array(), chain: "", txId: new Uint8Array(), asset: undefined };
}
exports.ConfirmTokenRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.txId.length !== 0) {
            writer.uint32(26).bytes(message.txId);
        }
        if (message.asset !== undefined) {
            types_2.Asset.encode(message.asset, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmTokenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.txId = reader.bytes();
                    break;
                case 4:
                    message.asset = types_2.Asset.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            txId: isSet(object.txId) ? bytesFromBase64(object.txId) : new Uint8Array(),
            asset: isSet(object.asset) ? types_2.Asset.fromJSON(object.asset) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.txId !== undefined &&
            (obj.txId = base64FromBytes(message.txId !== undefined ? message.txId : new Uint8Array()));
        message.asset !== undefined && (obj.asset = message.asset ? types_2.Asset.toJSON(message.asset) : undefined);
        return obj;
    },
    create(base) {
        return exports.ConfirmTokenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConfirmTokenRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.txId = object.txId ?? new Uint8Array();
        message.asset = (object.asset !== undefined && object.asset !== null) ? types_2.Asset.fromPartial(object.asset) : undefined;
        return message;
    },
};
function createBaseConfirmTokenResponse() {
    return {};
}
exports.ConfirmTokenResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmTokenResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.ConfirmTokenResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseConfirmTokenResponse();
        return message;
    },
};
function createBaseConfirmTransferKeyRequest() {
    return { sender: new Uint8Array(), chain: "", txId: new Uint8Array() };
}
exports.ConfirmTransferKeyRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.txId.length !== 0) {
            writer.uint32(26).bytes(message.txId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmTransferKeyRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.txId = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            txId: isSet(object.txId) ? bytesFromBase64(object.txId) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.txId !== undefined &&
            (obj.txId = base64FromBytes(message.txId !== undefined ? message.txId : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.ConfirmTransferKeyRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConfirmTransferKeyRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.txId = object.txId ?? new Uint8Array();
        return message;
    },
};
function createBaseConfirmTransferKeyResponse() {
    return {};
}
exports.ConfirmTransferKeyResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConfirmTransferKeyResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.ConfirmTransferKeyResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseConfirmTransferKeyResponse();
        return message;
    },
};
function createBaseLinkRequest() {
    return { sender: new Uint8Array(), chain: "", recipientAddr: "", asset: "", recipientChain: "" };
}
exports.LinkRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.recipientAddr !== "") {
            writer.uint32(26).string(message.recipientAddr);
        }
        if (message.asset !== "") {
            writer.uint32(34).string(message.asset);
        }
        if (message.recipientChain !== "") {
            writer.uint32(42).string(message.recipientChain);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLinkRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.recipientAddr = reader.string();
                    break;
                case 4:
                    message.asset = reader.string();
                    break;
                case 5:
                    message.recipientChain = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            recipientAddr: isSet(object.recipientAddr) ? String(object.recipientAddr) : "",
            asset: isSet(object.asset) ? String(object.asset) : "",
            recipientChain: isSet(object.recipientChain) ? String(object.recipientChain) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.recipientAddr !== undefined && (obj.recipientAddr = message.recipientAddr);
        message.asset !== undefined && (obj.asset = message.asset);
        message.recipientChain !== undefined && (obj.recipientChain = message.recipientChain);
        return obj;
    },
    create(base) {
        return exports.LinkRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseLinkRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.recipientAddr = object.recipientAddr ?? "";
        message.asset = object.asset ?? "";
        message.recipientChain = object.recipientChain ?? "";
        return message;
    },
};
function createBaseLinkResponse() {
    return { depositAddr: "" };
}
exports.LinkResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.depositAddr !== "") {
            writer.uint32(10).string(message.depositAddr);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLinkResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.depositAddr = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return { depositAddr: isSet(object.depositAddr) ? String(object.depositAddr) : "" };
    },
    toJSON(message) {
        const obj = {};
        message.depositAddr !== undefined && (obj.depositAddr = message.depositAddr);
        return obj;
    },
    create(base) {
        return exports.LinkResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseLinkResponse();
        message.depositAddr = object.depositAddr ?? "";
        return message;
    },
};
function createBaseCreateBurnTokensRequest() {
    return { sender: new Uint8Array(), chain: "" };
}
exports.CreateBurnTokensRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateBurnTokensRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        return obj;
    },
    create(base) {
        return exports.CreateBurnTokensRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateBurnTokensRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        return message;
    },
};
function createBaseCreateBurnTokensResponse() {
    return {};
}
exports.CreateBurnTokensResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateBurnTokensResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.CreateBurnTokensResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseCreateBurnTokensResponse();
        return message;
    },
};
function createBaseCreateDeployTokenRequest() {
    return {
        sender: new Uint8Array(),
        chain: "",
        asset: undefined,
        tokenDetails: undefined,
        address: new Uint8Array(),
        dailyMintLimit: "",
    };
}
exports.CreateDeployTokenRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.asset !== undefined) {
            types_2.Asset.encode(message.asset, writer.uint32(26).fork()).ldelim();
        }
        if (message.tokenDetails !== undefined) {
            types_2.TokenDetails.encode(message.tokenDetails, writer.uint32(34).fork()).ldelim();
        }
        if (message.address.length !== 0) {
            writer.uint32(50).bytes(message.address);
        }
        if (message.dailyMintLimit !== "") {
            writer.uint32(58).string(message.dailyMintLimit);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateDeployTokenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.asset = types_2.Asset.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.tokenDetails = types_2.TokenDetails.decode(reader, reader.uint32());
                    break;
                case 6:
                    message.address = reader.bytes();
                    break;
                case 7:
                    message.dailyMintLimit = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            asset: isSet(object.asset) ? types_2.Asset.fromJSON(object.asset) : undefined,
            tokenDetails: isSet(object.tokenDetails) ? types_2.TokenDetails.fromJSON(object.tokenDetails) : undefined,
            address: isSet(object.address) ? bytesFromBase64(object.address) : new Uint8Array(),
            dailyMintLimit: isSet(object.dailyMintLimit) ? String(object.dailyMintLimit) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.asset !== undefined && (obj.asset = message.asset ? types_2.Asset.toJSON(message.asset) : undefined);
        message.tokenDetails !== undefined &&
            (obj.tokenDetails = message.tokenDetails ? types_2.TokenDetails.toJSON(message.tokenDetails) : undefined);
        message.address !== undefined &&
            (obj.address = base64FromBytes(message.address !== undefined ? message.address : new Uint8Array()));
        message.dailyMintLimit !== undefined && (obj.dailyMintLimit = message.dailyMintLimit);
        return obj;
    },
    create(base) {
        return exports.CreateDeployTokenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateDeployTokenRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.asset = (object.asset !== undefined && object.asset !== null) ? types_2.Asset.fromPartial(object.asset) : undefined;
        message.tokenDetails = (object.tokenDetails !== undefined && object.tokenDetails !== null)
            ? types_2.TokenDetails.fromPartial(object.tokenDetails)
            : undefined;
        message.address = object.address ?? new Uint8Array();
        message.dailyMintLimit = object.dailyMintLimit ?? "";
        return message;
    },
};
function createBaseCreateDeployTokenResponse() {
    return {};
}
exports.CreateDeployTokenResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateDeployTokenResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.CreateDeployTokenResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseCreateDeployTokenResponse();
        return message;
    },
};
function createBaseCreatePendingTransfersRequest() {
    return { sender: new Uint8Array(), chain: "" };
}
exports.CreatePendingTransfersRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreatePendingTransfersRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        return obj;
    },
    create(base) {
        return exports.CreatePendingTransfersRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreatePendingTransfersRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        return message;
    },
};
function createBaseCreatePendingTransfersResponse() {
    return {};
}
exports.CreatePendingTransfersResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreatePendingTransfersResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.CreatePendingTransfersResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseCreatePendingTransfersResponse();
        return message;
    },
};
function createBaseCreateTransferOwnershipRequest() {
    return { sender: new Uint8Array(), chain: "", keyId: "" };
}
exports.CreateTransferOwnershipRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.keyId !== "") {
            writer.uint32(26).string(message.keyId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateTransferOwnershipRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.keyId = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            keyId: isSet(object.keyId) ? String(object.keyId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.keyId !== undefined && (obj.keyId = message.keyId);
        return obj;
    },
    create(base) {
        return exports.CreateTransferOwnershipRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateTransferOwnershipRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.keyId = object.keyId ?? "";
        return message;
    },
};
function createBaseCreateTransferOwnershipResponse() {
    return {};
}
exports.CreateTransferOwnershipResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateTransferOwnershipResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.CreateTransferOwnershipResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseCreateTransferOwnershipResponse();
        return message;
    },
};
function createBaseCreateTransferOperatorshipRequest() {
    return { sender: new Uint8Array(), chain: "", keyId: "" };
}
exports.CreateTransferOperatorshipRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.keyId !== "") {
            writer.uint32(26).string(message.keyId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateTransferOperatorshipRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.keyId = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            keyId: isSet(object.keyId) ? String(object.keyId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.keyId !== undefined && (obj.keyId = message.keyId);
        return obj;
    },
    create(base) {
        return exports.CreateTransferOperatorshipRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateTransferOperatorshipRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.keyId = object.keyId ?? "";
        return message;
    },
};
function createBaseCreateTransferOperatorshipResponse() {
    return {};
}
exports.CreateTransferOperatorshipResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateTransferOperatorshipResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.CreateTransferOperatorshipResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseCreateTransferOperatorshipResponse();
        return message;
    },
};
function createBaseSignCommandsRequest() {
    return { sender: new Uint8Array(), chain: "" };
}
exports.SignCommandsRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignCommandsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        return obj;
    },
    create(base) {
        return exports.SignCommandsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignCommandsRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        return message;
    },
};
function createBaseSignCommandsResponse() {
    return { batchedCommandsId: new Uint8Array(), commandCount: 0 };
}
exports.SignCommandsResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.batchedCommandsId.length !== 0) {
            writer.uint32(10).bytes(message.batchedCommandsId);
        }
        if (message.commandCount !== 0) {
            writer.uint32(16).uint32(message.commandCount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignCommandsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.batchedCommandsId = reader.bytes();
                    break;
                case 2:
                    message.commandCount = reader.uint32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            batchedCommandsId: isSet(object.batchedCommandsId) ? bytesFromBase64(object.batchedCommandsId) : new Uint8Array(),
            commandCount: isSet(object.commandCount) ? Number(object.commandCount) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        message.batchedCommandsId !== undefined &&
            (obj.batchedCommandsId = base64FromBytes(message.batchedCommandsId !== undefined ? message.batchedCommandsId : new Uint8Array()));
        message.commandCount !== undefined && (obj.commandCount = Math.round(message.commandCount));
        return obj;
    },
    create(base) {
        return exports.SignCommandsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignCommandsResponse();
        message.batchedCommandsId = object.batchedCommandsId ?? new Uint8Array();
        message.commandCount = object.commandCount ?? 0;
        return message;
    },
};
function createBaseAddChainRequest() {
    return { sender: new Uint8Array(), name: "", keyType: 0, params: new Uint8Array() };
}
exports.AddChainRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.name !== "") {
            writer.uint32(18).string(message.name);
        }
        if (message.keyType !== 0) {
            writer.uint32(32).int32(message.keyType);
        }
        if (message.params.length !== 0) {
            writer.uint32(42).bytes(message.params);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddChainRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.name = reader.string();
                    break;
                case 4:
                    message.keyType = reader.int32();
                    break;
                case 5:
                    message.params = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            name: isSet(object.name) ? String(object.name) : "",
            keyType: isSet(object.keyType) ? (0, types_1.keyTypeFromJSON)(object.keyType) : 0,
            params: isSet(object.params) ? bytesFromBase64(object.params) : new Uint8Array(),
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.name !== undefined && (obj.name = message.name);
        message.keyType !== undefined && (obj.keyType = (0, types_1.keyTypeToJSON)(message.keyType));
        message.params !== undefined &&
            (obj.params = base64FromBytes(message.params !== undefined ? message.params : new Uint8Array()));
        return obj;
    },
    create(base) {
        return exports.AddChainRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddChainRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.name = object.name ?? "";
        message.keyType = object.keyType ?? 0;
        message.params = object.params ?? new Uint8Array();
        return message;
    },
};
function createBaseAddChainResponse() {
    return {};
}
exports.AddChainResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddChainResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.AddChainResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseAddChainResponse();
        return message;
    },
};
function createBaseRetryFailedEventRequest() {
    return { sender: new Uint8Array(), chain: "", eventId: "" };
}
exports.RetryFailedEventRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sender.length !== 0) {
            writer.uint32(10).bytes(message.sender);
        }
        if (message.chain !== "") {
            writer.uint32(18).string(message.chain);
        }
        if (message.eventId !== "") {
            writer.uint32(26).string(message.eventId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRetryFailedEventRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.sender = reader.bytes();
                    break;
                case 2:
                    message.chain = reader.string();
                    break;
                case 3:
                    message.eventId = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            sender: isSet(object.sender) ? bytesFromBase64(object.sender) : new Uint8Array(),
            chain: isSet(object.chain) ? String(object.chain) : "",
            eventId: isSet(object.eventId) ? String(object.eventId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.sender !== undefined &&
            (obj.sender = base64FromBytes(message.sender !== undefined ? message.sender : new Uint8Array()));
        message.chain !== undefined && (obj.chain = message.chain);
        message.eventId !== undefined && (obj.eventId = message.eventId);
        return obj;
    },
    create(base) {
        return exports.RetryFailedEventRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRetryFailedEventRequest();
        message.sender = object.sender ?? new Uint8Array();
        message.chain = object.chain ?? "";
        message.eventId = object.eventId ?? "";
        return message;
    },
};
function createBaseRetryFailedEventResponse() {
    return {};
}
exports.RetryFailedEventResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRetryFailedEventResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.RetryFailedEventResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseRetryFailedEventResponse();
        return message;
    },
};
var tsProtoGlobalThis = (() => {
    if (typeof globalThis !== "undefined") {
        return globalThis;
    }
    if (typeof self !== "undefined") {
        return self;
    }
    if (typeof window !== "undefined") {
        return window;
    }
    if (typeof global !== "undefined") {
        return global;
    }
    throw "Unable to locate global object";
})();
function bytesFromBase64(b64) {
    if (tsProtoGlobalThis.Buffer) {
        return Uint8Array.from(tsProtoGlobalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = tsProtoGlobalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (tsProtoGlobalThis.Buffer) {
        return tsProtoGlobalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(String.fromCharCode(byte));
        });
        return tsProtoGlobalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
