// @axelar-network/axelarjs-types/build/module/axelar/nexus/v1beta1/params.js
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { Threshold } from "../../utils/v1beta1/threshold";
export const protobufPackage = "axelar.nexus.v1beta1";
function createBaseParams() {
    return {
        chainActivationThreshold: undefined,
        chainMaintainerMissingVoteThreshold: undefined,
        chainMaintainerIncorrectVoteThreshold: undefined,
        chainMaintainerCheckWindow: 0,
    };
}
export const Params = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.chainActivationThreshold !== undefined) {
            Threshold.encode(message.chainActivationThreshold, writer.uint32(10).fork()).ldelim();
        }
        if (message.chainMaintainerMissingVoteThreshold !== undefined) {
            Threshold.encode(message.chainMaintainerMissingVoteThreshold, writer.uint32(18).fork()).ldelim();
        }
        if (message.chainMaintainerIncorrectVoteThreshold !== undefined) {
            Threshold.encode(message.chainMaintainerIncorrectVoteThreshold, writer.uint32(26).fork()).ldelim();
        }
        if (message.chainMaintainerCheckWindow !== 0) {
            writer.uint32(32).int32(message.chainMaintainerCheckWindow);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseParams();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.chainActivationThreshold = Threshold.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.chainMaintainerMissingVoteThreshold = Threshold.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.chainMaintainerIncorrectVoteThreshold = Threshold.decode(reader, reader.uint32());
                    break;
                case 4:
                    message.chainMaintainerCheckWindow = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            chainActivationThreshold: isSet(object.chainActivationThreshold)
                ? Threshold.fromJSON(object.chainActivationThreshold)
                : undefined,
            chainMaintainerMissingVoteThreshold: isSet(object.chainMaintainerMissingVoteThreshold)
                ? Threshold.fromJSON(object.chainMaintainerMissingVoteThreshold)
                : undefined,
            chainMaintainerIncorrectVoteThreshold: isSet(object.chainMaintainerIncorrectVoteThreshold)
                ? Threshold.fromJSON(object.chainMaintainerIncorrectVoteThreshold)
                : undefined,
            chainMaintainerCheckWindow: isSet(object.chainMaintainerCheckWindow)
                ? Number(object.chainMaintainerCheckWindow)
                : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        message.chainActivationThreshold !== undefined && (obj.chainActivationThreshold = message.chainActivationThreshold
            ? Threshold.toJSON(message.chainActivationThreshold)
            : undefined);
        message.chainMaintainerMissingVoteThreshold !== undefined &&
            (obj.chainMaintainerMissingVoteThreshold = message.chainMaintainerMissingVoteThreshold
                ? Threshold.toJSON(message.chainMaintainerMissingVoteThreshold)
                : undefined);
        message.chainMaintainerIncorrectVoteThreshold !== undefined &&
            (obj.chainMaintainerIncorrectVoteThreshold = message.chainMaintainerIncorrectVoteThreshold
                ? Threshold.toJSON(message.chainMaintainerIncorrectVoteThreshold)
                : undefined);
        message.chainMaintainerCheckWindow !== undefined &&
            (obj.chainMaintainerCheckWindow = Math.round(message.chainMaintainerCheckWindow));
        return obj;
    },
    create(base) {
        return Params.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseParams();
        message.chainActivationThreshold =
            (object.chainActivationThreshold !== undefined && object.chainActivationThreshold !== null)
                ? Threshold.fromPartial(object.chainActivationThreshold)
                : undefined;
        message.chainMaintainerMissingVoteThreshold =
            (object.chainMaintainerMissingVoteThreshold !== undefined && object.chainMaintainerMissingVoteThreshold !== null)
                ? Threshold.fromPartial(object.chainMaintainerMissingVoteThreshold)
                : undefined;
        message.chainMaintainerIncorrectVoteThreshold =
            (object.chainMaintainerIncorrectVoteThreshold !== undefined &&
                object.chainMaintainerIncorrectVoteThreshold !== null)
                ? Threshold.fromPartial(object.chainMaintainerIncorrectVoteThreshold)
                : undefined;
        message.chainMaintainerCheckWindow = object.chainMaintainerCheckWindow ?? 0;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
